src/Tester/SerializationTests/DeepCopyTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Orleans.Serialization;
+using Orleans.UnitTest.GrainInterfaces;
+using UnitTests.GrainInterfaces;
+
+namespace UnitTests.Serialization
+{
+    /// <summary>
+    /// Test the deep copy of built-in and user-defined types
+    /// </summary>
+    [TestClass]
+    public class DeepCopyTests
+    {
+
+        [TestInitialize]
+        public void InitializeForTesting()
+        {
+            SerializationManager.InitializeForTesting();
+        }
+
+
+        [TestMethod, TestCategory("BVT"), TestCategory("Functional"), TestCategory("Serialization")]
+        public void DeepCopyTests_BuiltinCollections()
+        {
+            {
+                var original = new int[] { 0, 1, 2 };
+                var copy = (int[])SerializationManager.DeepCopy(original);
+                copy[2] = 0;
+                Assert.AreEqual(original[2], 2);
```

Comment:
probably worth to ALSO do a `CollectionAssert.AreEquivalent(original, copy)`. Unless this is covered somewhere else?